eo-maven-plugin/src/main/java/org/eolang/maven/Dependencies.java

```diff
@@ -0,0 +1,178 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.jcabi.log.Logger;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import javax.json.JsonValue;
+import org.apache.maven.model.Dependency;
+
+/**
+ * Maven dependencies abstraction.
+ *
+ * @since 0.28.11
+ */
+interface Dependencies {
+
+    /**
+     * Converts to a plain list.
+     *
+     * @return List of Maven Dependencies
+     */
+    List<Dependency> toList();
+
+    /**
+     * Filtered dependencies.
+     *
+     * @since 0.28.11
+     */
+    class FilteredDependencies implements Dependencies {
+
+        /**
+         * Decorated.
+         */
+        private final Dependencies dependencies;
+
+        /**
+         * All filters.
+         */
+        private final Collection<Predicate<Dependency>> filters;
+
+        /**
+         * The main constructor.
+         *
+         * @param dependencies Decorated
+         * @param filters All fiters
+         */
+        FilteredDependencies(
+            final Dependencies dependencies,
+            final Collection<Predicate<Dependency>> filters
+        ) {
+            this.dependencies = dependencies;
+            this.filters = filters;
+        }
+
+        @Override
+        public List<Dependency> toList() {
+            return this.dependencies.toList()
+                .stream()
+                .filter(this::applyAllFilters)
+                .collect(Collectors.toList());
+        }
+
+        /**
+         * Apply all filters for Dependency.
+         *
+         * @param dependency Dependency
+         * @return True if all filters were passed
+         */
+        private boolean applyAllFilters(final Dependency dependency) {
```

Comment:
@volodya-lombrozo Can we rename this method? Like apply, match, test or something else. According to this: https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html

Reply:
@Graur Fixed. Please, take a look.